app/src/@anchor-protocol/app-provider/queries/earn/useProjectedEarnApy.ts

```diff
@@ -0,0 +1,124 @@
+import { moneyMarket } from '@anchor-protocol/types';
+import { terraNativeBalancesQuery } from '@libs/app-fns';
+
+interface ProjectedEarnApyWasmQuery {
+  overseerDynRateState: WasmQuery<
+    moneyMarket.overseer.DynRateState,
+    moneyMarket.overseer.DynRateStateResponse
+  >;
+  overseerConfig: WasmQuery<
+    moneyMarket.overseer.Config,
+    moneyMarket.overseer.ConfigResponse
+  >;
+}
+
+// the logic is copied from https://github.com/Anchor-Protocol/money-market-contracts/blob/5d902c3f029eeda6597e54763e4b608a716603a1/contracts/overseer/src/contract.rs#L389
+const projectedEarnApyQuery = async (
+  blocksPerYear: number,
+  overseerContract: HumanAddr,
+  queryClient: QueryClient,
+) => {
+  const { uUST } = await terraNativeBalancesQuery(
+    overseerContract,
+    queryClient,
+  );
+  const currentYieldReserve = big(uUST);
+
+  const { overseerDynRateState, overseerConfig } =
+    await wasmFetch<ProjectedEarnApyWasmQuery>({
+      ...queryClient,
+      id: 'projected-earn-apy',
+      wasmQuery: {
+        overseerDynRateState: {
+          contractAddress: overseerContract,
+          query: { dynrate_state: {} },
+        },
+        overseerConfig: {
+          contractAddress: overseerContract,
+          query: { config: {} },
+        },
+      },
+    });
+  const prevYieldReserve = big(overseerDynRateState.prev_yield_reserve);
+  const dynRateMaxChange = big(overseerConfig.dyn_rate_maxchange);
+  const thresholdDepositRate = big(overseerConfig.threshold_deposit_rate);
+  const dynRateYearIncreaseExpectation = big(
+    overseerConfig.dyn_rate_yr_increase_expectation,
+  );
+  const dynApyMin = big(overseerConfig.dyn_rate_min).mul(blocksPerYear);
+  const dynApyMax = big(overseerConfig.dyn_rate_max).mul(blocksPerYear);
+
+  const currentRate = thresholdDepositRate.mul(blocksPerYear);
+
+  let isCurrentYieldReserveHigher = currentYieldReserve > prevYieldReserve;
+
+  const yieldReserveDelta = isCurrentYieldReserveHigher
+    ? currentYieldReserve.minus(prevYieldReserve)
+    : prevYieldReserve.minus(currentYieldReserve);
+
+  let yieldReserveChange = prevYieldReserve.eq(0)
+    ? big(1)
+    : yieldReserveDelta.div(prevYieldReserve);
+
+  let increaseExpectation = dynRateYearIncreaseExpectation;
+  if (!isCurrentYieldReserveHigher) {
+    yieldReserveChange = yieldReserveChange.add(increaseExpectation);
+  } else if (currentRate.lt(yieldReserveChange)) {
+    yieldReserveChange = yieldReserveChange.minus(increaseExpectation);
+  } else {
+    isCurrentYieldReserveHigher = !isCurrentYieldReserveHigher;
+    increaseExpectation = increaseExpectation.minus(yieldReserveChange);
+  }
+
+  if (yieldReserveChange.lt(dynRateMaxChange)) {
+    yieldReserveChange = dynRateMaxChange;
+  }
+
+  let projectedEarnApy = big(0);
+  if (isCurrentYieldReserveHigher) {
+    projectedEarnApy = currentRate.plus(yieldReserveChange);
+  } else if (currentRate.lt(yieldReserveChange)) {
```

Comment:
@RodionChachura this is the issue here, this should be currentRate.gt(...) not .lt()

Reply:
@RodionChachura also see if you can split this out into smaller functions to improve readability, ie,

const { change, isHigher } = computeYieldReserveChange(...)

const projected = computeProjectedApr(...)

Reply:
🤦‍♂️✅ https://github.com/Anchor-Protocol/anchor-web-app/pull/309/commits/678de5c365b9ec42aff264447845a4a1320a4cbe
![image](https://user-images.githubusercontent.com/17750556/161179539-7cf6d92e-eb1a-420f-9864-5e5031d76d1f.png)

Reply:
As a user I'm not really sure through the UI what the "Projected 19.97%" means here. 
I think we should add a tooltip explaining what the projection means.

Reply:
As a user I'm not really sure through the UI what the "Projected 19.97%" means here. 
I think we should add a tooltip explaining what the projection means.